### ISSUE TITLE
Include as little JS as possible prior to running tests

### ISSUE THREAD

**Issue by foolip** [foolip/mdn-bcd-collector#407]
This dependency was added in 8637e29b4eb3d690dfe6d07b9129f5fc0611e423 with the comment "Include core-js polyfill for Promise/Set/etc. support in older browsers".

This is in line with our use of JSON3, but adds more stuff to the global. Any generated tests for Promise/Set/etc would be invalidated by this, which is also the case for JSON3. But JSON3 has a `JSON3.noConflict()` which we could use to avoid this.

We should have as little JS as possible run before the actual tests, and once they have finished, we can pull in limitless amounts of code and stylesheets to present the results in a beautiful way :)

**Comment by vinyldarkscratch** [comment]
I wonder if there's a better, smaller option for polyfilling the promises; that would help out greatly.  Unfortunately, promises are a core component in the functionality of the Worker and ServiceWorker tests -- at least, in a reasonable way.  Alas, it would certainly cause issues for automated JavaScript feature testing if we're to add that...

**Comment by foolip** [comment]
Can you point to the code where we depend on promises? It's probably straightforward to replace it with something relatively antique to avoid the issue.

**Comment by vinyldarkscratch** [comment]
We're using it to handle the messages back from both the Workers.  For the service workers, it's not as much of a problem since Promises were natively handled before Service Workers, but regular Web Workers were supported first.

**Comment by foolip** [comment]
I see, so it's used to wait for all tests to finish. It will be a little bit of bookkeeping, but if that's all then it can be rewritten to check if all tests are finished every time one test finishes. It could also simplify things to never do any work in parallel, but I don't know if that's an important performance optimization?

**Comment by foolip** [comment]
For the use of promises to await results from many worker tests, I believe https://github.com/foolip/mdn-bcd-collector/issues/436 will also help. The worker script would run all the tests and post back results in a single message, so there wouldn't be any need for the bookkeeping that `Promise.allSettled` now simplifies.

**Comment by vinyldarkscratch** [comment]
I've just implemented the code to send all the tests at once, so promises are no longer needed, and thus `core-js` is no longer a dependency!

**Comment by foolip** [comment]
Excellent, thanks! Let's leave this open though to see if there's more we can get rid of when things have settled down.

**Comment by queengooborg** [comment]
With #1690 now merged, the only pre-test dependency is our own `harness.js`, so I think it's safe to close this issue.  I also plan to minify `harness.js` which should help even more.
